### PR TITLE
Add more validations upfront (k8s and postgres)

### PR DESCRIPTION
**Pull request by nvthongswansea**
Ref #150 . Changes:
- Add more validations upfront (k8s and postgres)

### CODE REVIEW

File: gridscale/resource_gridscale_postgresql.go [hunk 1]
```diff
@@ -132,6 +132,12 @@ func resourceGridscalePostgreSQL() *schema.Resource {
 				Description: "Maximum CPU core count. The PostgreSQL instance's CPU core count will be autoscaled based on the workload. The number of cores stays between 1 and `max_core_count`.",
 				Optional:    true,
 				Computed:    true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					if 1 > v.(int) || v.(int) > 32 {
+						errors = append(errors, fmt.Errorf("%v is not a valid value for number of \"max_core_count\". Valid value should be between 1 and 32\n", v.(int)))
+					}
+					return
+				},
```
Comment (bkircher) [inline gridscale/resource_gridscale_postgresql.go:140]: Are you sure this is correct? I think this needs to be 

```go
func(v interface{}, k string) (ws []string, errors []error) []error {
  // …
  return errors
}
```

Here and other ValidateFunc values above.

**Review by bkircher — CHANGES_REQUESTED** [review]

File: gridscale/resource_gridscale_postgresql.go (hunk 1 above)
Comment (nvthongswansea) [reply gridscale/resource_gridscale_postgresql.go:140]: @bkircher fyi, 
image
image

Comment (nvthongswansea) [reply gridscale/resource_gridscale_postgresql.go:140]: and [Named return values](https://tour.golang.org/basics/7)

Comment (bkircher) [reply gridscale/resource_gridscale_postgresql.go:140]: :sunglasses: Ahh, learned something new here. Thanks!

**Review by bkircher — APPROVED** [review]

**Pull request merged**